src/axom/primal/operators/evaluate_integral.hpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ * \file evaluate_integral.hpp
+ *
+ * \brief Consists of methods that evaluate integrals over regions defined
+ * by Bezier curves, such as 2D area integrals and scalar/vector field line integrals
+ *
+ * Line integrals are computed with simple 1D Gaussian Quadrature using rules generated
+ * by MFEM. 2D area integrals computed with "Spectral Mesh-Free Quadrature for Planar 
+ * Regions Bounded by Rational Parametric Curves" by David Gunderman et al.
+ */
+
+#ifndef PRIMAL_EVAL_INTEGRAL_HPP_
+#define PRIMAL_EVAL_INTEGRAL_HPP_
+
+// Axom includes
+#include "axom/config.hpp"  // for compile-time configuration options
+#include "axom/primal.hpp"
+
+#include "axom/primal/operators/detail/evaluate_integral_impl.hpp"
+
+// MFEM includes
+#ifdef AXOM_USE_MFEM
+  #include "mfem.hpp"
+#else
+  #error "Primal's integral evaluation functions require mfem library."
+#endif
+
+// C++ includes
+#include <cmath>
+
+// Create type definitions the different field types (vector vs scalar)
+//   to allow for proper overloading of line integrals
+using Point2D = axom::primal::Point<double, 2>;
+using Vector2D = axom::primal::Vector<double, 2>;
+
+namespace axom
+{
+namespace primal
+{
+/*!
+ * \brief Evaluate a line integral along a collection of Bezier curves
+ *
+ * The curve need not be connected, simply adding the portion along each segment.
+ * Uses Gaussian quadrature generated by MFEM.
+ *
+ * \param [in] cs the array of Bezier curve objects
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input, and return a double (scalar field) or 
+ * 2D vector object (vector field).
+ * \param [in] npts the number of Gaussian quadrature nodes for each component
+ * \return the value of the integral
+ */
+template <class Lambda>
+double evaluate_line_integral(const axom::Array<primal::BezierCurve<double, 2>>& cs,
+                              Lambda&& integrand,
+                              int npts)
+{
+  // Generate quadrature library, defaulting to GaussLegendre quadrature.
+  //  Use the same one for every curve in the polygon
+  //  Quadrature order is equal to 2*N - 1
+  static mfem::IntegrationRules my_IntRules(mfem::Quadrature1D::GaussLegendre);
+  const mfem::IntegrationRule* quad =
+    &(my_IntRules.Get(mfem::Geometry::SEGMENT, 2 * npts - 1));
+
+  double total_integral = 0.0;
+  for(int i = 0; i < cs.size(); i++)
+  {
+    // Compute the line integral along each component
+    total_integral +=
+      detail::evaluate_line_integral_component(cs[i], integrand, quad);
+  }
```

Comment:
Minor: This could use the range-for pattern:

```suggestion
  for(const auto& curve : cs)
  {
    // Compute the line integral along each component
    total_integral +=
      detail::evaluate_line_integral_component(curve, integrand, quad);
  }
```